test/topology_custom/test_replace_with_same_ip_twice.py

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import time
+from test.pylib.scylla_cluster import ReplaceConfig
+from test.pylib.manager_client import ManagerClient
+from test.topology.util import wait_for_token_ring_and_group0_consistency
+from test.pylib.internal_types import ServerInfo
+import pytest
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_replace_with_same_ip_twice(manager: ManagerClient) -> None:
+    """Replace an existing node with new node using the same IP address, and then change its IP"""
+
+    logger.info("starting a cluster with two nodes")
+    servers = await manager.servers_add(3, config={'failure_detector_timeout_in_ms': 2000})
+    logger.info(f"cluster started {servers}")
+
+    async def replace_with_same_ip(s: ServerInfo) -> ServerInfo:
+        logger.info(f"stopping server {s.server_id}")
+        await manager.server_stop(s.server_id)
+
+        logger.info(f"replacing server {s.server_id} with same ip")
+        replace_cfg = ReplaceConfig(replaced_id = s.server_id, reuse_ip_addr = True, use_host_id = False)
+        result: ServerInfo = await manager.server_add(replace_cfg)
```

Comment:
By the moment you call this, the failure detector might not yet have marked the stopped server as DOWN. IIRC we have a check that we cannot replace an alive node. In that case I wonder how the test can be passing -- 2 seconds (`failure_detector_timeout_in_ms` above) should not be passing so quick compared to how server_add runs

Reply:
> That's because of [this feature](https://github.com/scylladb/scylladb/pull/15926/commits/9775b1c12dc5c31f9a6d87f071f1d54b323d5edf)

Ah, makes sense

> BTW, don't you know why the [merge commit](https://github.com/scylladb/scylladb/commit/03ecc8457c8535e78daac6e4317978a74439faa8) doesn't appear in the history of scylla_cluster.py

See man git-log, `--diff-merges` option. Default is `off`, you might want `m`
(`git log -m <file>` or `git log -m -p <file>` if you want to see the changes)